### PR TITLE
Validate generated schemas in generator script

### PR DESCRIPTION
**Pull request by verbanicm**
**Related:** #6371 

**Description**

Moved validation logic from [main_test.go](https://github.com/GoogleContainerTools/skaffold/blob/main/hack/schemas/main_test.go) into main file for immediate feedback on invalid schema generation. 

This should improve reliability when updating schema changes and prevent unwanted breakages in docs

**Note** v2beta19 config needed updated to allow new logic and unit tests to pass. This should not have any user facing changes.

### CODE REVIEW

**Review by gsquared94 — APPROVED** [review]
lgtm

File: hack/schemas/main.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ import (
 	"strings"
 	"sync"
 
+	"github.com/xeipuuv/gojsonschema"
+
```
Comment (gsquared94) [inline hack/schemas/main.go:35]: I think this new line on 35 is the cause of your `linter` error
https://app.travis-ci.com/github/GoogleContainerTools/skaffold/jobs/530055237#L316

Comment (gsquared94) [reply hack/schemas/main.go:35]:
```
import (
  // stdlib

  // third-party

  // other packages of that project
)
```

https://stackoverflow.com/a/59964885/4540147

Comment (verbanicm) [reply hack/schemas/main.go]: Thanks!!!

**Comment by verbanicm** [comment]
[CI check failure](https://app.travis-ci.com/github/GoogleContainerTools/skaffold/jobs/530072853) is expected due to modifying existing schema.

**Pull request merged**